src/components/Tree/TreeTitle.tsx

```diff
@@ -0,0 +1,80 @@
+import * as _ from 'lodash'
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+
+import { UIComponent, childrenExist, createShorthandFactory } from '../../lib'
+import { treeTitleBehavior } from '../../lib/accessibility'
+import { Accessibility } from '../../lib/accessibility/types'
+import {
+  commonUIComponentPropTypes,
+  childrenComponentPropTypes,
+  contentComponentPropsTypes,
+} from '../../lib/commonPropTypes'
+import {
+  UIComponentProps,
+  ChildrenComponentProps,
+  ContentComponentProps,
+} from '../../lib/commonPropInterfaces'
+
+export interface TreeTitleProps
+  extends UIComponentProps<any, any>,
+    ChildrenComponentProps,
+    ContentComponentProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   * @default treeTitleBehavior
+   */
+  accessibility?: Accessibility
+
+  /** Whether or not the subtree of the item is in the open state. */
+  open?: boolean
+
+  /** Whether or not the item has a subtree. */
+  hasSubtree?: boolean
+}
+
+class TreeTitle extends UIComponent<TreeTitleProps> {
+  static create: Function
+
+  static className = 'ui-tree__title'
+
+  static displayName = 'TreeTitle'
+
+  static propTypes = {
+    ...commonUIComponentPropTypes,
+    ...childrenComponentPropTypes,
+    ...contentComponentPropsTypes,
+    accessibility: PropTypes.func,
+    open: PropTypes.bool,
+    hasSubtree: PropTypes.bool,
+  }
+
+  public static defaultProps = {
+    as: 'a',
+    href: '#',
```

Comment:
agree with @layershifter's comment on `href` default value. In the future PR for #511 can you please remove this default value and add keyboard handlers for `Enter` and `Space`?